### ISSUE TITLE
Ship logs to the log group /buildkite/elastic-stack-init even if the (custom) bootstrap script fails

### ISSUE THREAD

**Issue by zsims** [buildkite/elastic-ci-stack-for-aws#450]
We've found when specifiying a custom boostrap script (via the `BootstrapScriptUrl`) parameter, logs will not be written to the CloudWatch the `/buildkite/elastic-stack-init` log group if the bootstrap script fails.

E.g. with a custom bootstrap script:
```
#!/usr/bin/env bash
echo "I won't be in the logs"
exit 1
```

Then you'll never see "I won't be in the logs"

**Comment by zsims** [comment]
Maybe flushing the logs in a bash trap? Or at least if the custom bootstrap script fails?

I couldn't find anything about flushing the logs in https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/AgentReference.html though... If that turns out to be a dead end, then perhaps this could be an explicit log to AWS CW Logs?

**Comment by zsims** [comment]
I'll raise a ticket with AWS and see if there's a sane way to do this. Otherwise will have a crack at something custom.

**Comment by zsims** [comment]
AWS replied that the official AWS EC2 agent will handle this scenario (and will flush on service launch/exit). However, Elastic CI is using https://github.com/sj26/journald-cloudwatch-logs -- will look into how a flush could be forced.

**Comment by lox** [comment]
The official awslogs agent doesn't support systemd / journald unfortunately.

**Comment by zsims** [comment]
Good point :(. I've raised PR #492, as the agent should start a lot sooner than that bootstrap script. Still testing it, but it makes a lot of sense in theory 😁
